eclipse/winery
# [WIP]Feature/prepare yaml roundtrip tests

## Improve converter

- Add converter junit5 test cases
- Add derived_from type to ArtifactTypes for round trip
  - either implementation or deployment
- Add WineryPropertiesDefinition while converting from yaml to xml
- Fix NodeTypeImplementation Builder
- Add some comments
- Add some @nullable and @nonnull annotations
- Replace empty exception catches with error logs


- [x] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request. Especially, we require **a single commit**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [x] Ensure that you appear in `NOTICE` at Copyright Holders
- [x] Tests created for changes
- [ ] Screenshots added (for UI changes)
